docs/stdlib/constraints.rst

```diff
@@ -35,9 +35,8 @@ Constraints
 
 .. eql:constraint:: std::one_of(variadic members: anytype)
 
-    Specifies the list of allowed values directly.
-
-    Example:
+    Specifies a list of :eql:type:`anytype` as one allowed value of
```

Comment:
```suggestion
    Specifies a list of allowed values.
    
    Example:
```

This almost reads to me as if I should provide a list of permitted types. I think noting `anytype` in the signature is sufficient here to convey that aspect of the constraint.

Reply:
Go ahead and apply this guidance globally. I don't think the one-liner is the right place to expose the type restrictions of arguments and the like. I think we can leave that info in the signature.

Reply:
So we want to remove `anytype` references because the signature should concisely show this to the user?

Reply:
> ```suggestion
>     Specifies a list of allowed values.
>     
>     Example:
> ```
> 
> This almost reads to me as if I should provide a list of permitted types. I think noting `anytype` in the signature is sufficient here to convey that aspect of the constraint.

I'm not quite following. "Allowed values" is vague if that's the only thing read. I believe there's a higher likelihood of users reading the description over the signature. When read, you might go "what's allowed? Only values from standard library types?"

Reply:
Yes, this is somewhat vague, but the signature and the detailed description are there to fill in any gaps the one-liner leaves.

I thought about recommending we move that out into a detail paragraph after the one-liner, but I don't think it's necessary for this. If it had a type restriction, it might make sense to call that out in prose, but since it doesn't, I don't see a lot of value in it.

Reply:
Looking at `anytype`'s original description, it reads this:

> Generic type.
>
> It is a placeholder used in cases where no specific type requirements are needed, such as defining polymorphic parameters in functions and operators.

In our PR, we've changed to this.

> Represents a generic type.
>
> ... specific type requirements are needed, or are not necessary, ...

A suggestion from @vpetrovykh was to reword the amended section to read out as:

> ... where there aren't specific type requirements, such as ...

If we *did* have specific type requirements, this means we reference `str` or other builtin types to my knowledge. Having this referenced seems useful to those who want to know a specific type is not required, aka. meaning it may be universally applied to all known types. We could remove this, but then we'd need to clarify our proposal to be "Specifies a list of any type value."